tapdb/assets_store.go

```diff
@@ -3526,6 +3548,90 @@ func (a *AssetStore) TxHeight(ctx context.Context, txid chainhash.Hash) (uint32,
 	return uint32(dbBlockHeight), nil
 }
 
+// InsertBurn inserts a burnt asset to the db which includes all related
+// metadata.
+func (a *AssetStore) InsertBurn(ctx context.Context, txid []byte, note string,
+	assetID, groupKey []byte, amount uint64) error {
+
+	var writeTxOpts AssetStoreTxOptions
+	return a.db.ExecTx(ctx, &writeTxOpts, func(q ActiveAssetsStore) error {
+		// Look up the transfer that relates to this txid.
+		// TODO(george): We assume there's a single burn in every txn,
+		// this will need to be fixed in the future.
+		assetTransfers, err := q.QueryAssetTransfers(ctx, TransferQuery{
+			AnchorTxHash: txid,
+		})
+		if err != nil {
+			return fmt.Errorf("unable to query asset transfers: %w",
+				err)
+		}
+		assetTransfer := assetTransfers[0]
```

Comment:
Should check that we have exactly one result (which should always be the case when we query with an anchor TX hash).

Reply:
Yeah otherwise, we'll panic here.